app/uk/gov/hmrc/merchandiseinbaggage/repositories/DeclarationRepository.scala

```diff
@@ -95,21 +103,29 @@ class DeclarationRepositoryImpl @Inject()(mongo: () => DB)(implicit ec: Executio
         case None => Json.obj("mibReference" -> mibReference.value)
       }
 
-    collection.find(query, None).one[Declaration]
+    collection.find(query, None).one[Declaration].map(_.map(decryptDeclaration))
   }
 
   override def findLatestBySessionId(sessionId: SessionId): Future[Declaration] = {
     val query: (String, JsValueWrapper) = s"${Declaration.sessionId}" -> JsString(sessionId.value)
-    find(query).map(latest)
+    find(query).map(latest).map(decryptDeclaration)
   }
 
-  override def findAll: Future[List[Declaration]] = super.findAll()
+  override def findAll: Future[List[Declaration]] = super.findAll().map(_.map(decryptDeclaration))
 
   //TODO do we want to take some measure to stop getting called in prod!? Despite being in protected zone
   override def deleteAll(): Future[Unit] = super.removeAll().map(_ => ())
 
-  override def findBy(mibReference: MibReference, eori: Eori): Future[Option[Declaration]] = {
-    val query: Seq[(String, JsValueWrapper)] = Seq(("mibReference", JsString(mibReference.value)), ("eori.value", JsString(eori.value)))
-    find(query: _*).map(_.headOption)
+  def findBy(mibReference: MibReference, eori: Eori): Future[Option[Declaration]] = {
+    val encryptedEori: Eori = encryptEori(eori)
+    val query = Json.obj(
+      "mibReference" -> mibReference.value,
+      "eori.value"   -> Json.obj("$in" -> Json.arr(eori.value, encryptedEori.value))
+    )
+    collection.find(query, None).one[Declaration].map {
+      case Some(declaration) if eori == declaration.eori => Some(declaration)
+      case Some(declaration)                             => Some(decryptDeclaration(declaration))
+      case None                                          => None
+    }
```

Comment:
can we remove above `map` and use `.map(_.map(decryptDeclaration))` instead, won't it work ?

Reply:
if a match is found using a plain text eori then the declaration is no encrypted and decrypt is not needed. Have changed it to check the .encrypted flat to make it look more obvious.